File: CS/NewFieldsForm.cs
using System.Windows.Forms;

namespace RichEditMailMergeTemplateCreator {
    public partial class NewFieldsForm : Form {
        public NewFieldsForm() {
            InitializeComponent();
        }

        public string[] FieldNames {
            get {
                return memoEdit1.Lines;
            }
            set {
                memoEdit1.Lines = value;
            }
        }
    }
}

File: VB/NewFieldsForm.vb
Imports System.Windows.Forms

Namespace RichEditMailMergeTemplateCreator
    Partial Public Class NewFieldsForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Public Property FieldNames() As String()
            Get
                Return memoEdit1.Lines
            End Get
            Set(ByVal value As String())
                memoEdit1.Lines = value
            End Set
        End Property
    End Class
End Namespace
